modules/xmpp/ChatRoom.js

```diff
@@ -430,6 +432,29 @@ export default class ChatRoom extends Listenable {
 
         // process nodes to extract data needed for MUC_JOINED and
         // MUC_MEMBER_JOINED events
+        const extractIdentityInformation = node => {
+            const identity = {};
+            const userInfo = node.children.find(c => c.tagName === 'user');
+
+            if (userInfo) {
+                identity.user = {};
+                for (const tag of [ 'id', 'name', 'avatar' ]) {
+                    const child
+                        = userInfo.children.find(c => c.tagName === tag);
+
+                    if (child) {
+                        identity.user[tag] = child.value;
+                    }
+                }
+            }
+            const groupInfo = node.children.find(c => c.tagName === 'group');
+
+            if (groupInfo) {
+                identity.group = groupInfo.value;
+            }
+
+            return identity;
+        };
```

Comment:
I'm worried that having this inline causes more confusion to an already fairly lengthy function. If this piece is re-usable, would it be possible to move this out on its own? Maybe some xml parsing utils somewhere?